[2024-04-29T14:04:52.485+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Units_statistics_DAG.ETL_job scheduled__2024-04-29T13:40:00+00:00 map_index=16 [queued]>
[2024-04-29T14:04:52.502+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Units_statistics_DAG.ETL_job scheduled__2024-04-29T13:40:00+00:00 map_index=16 [queued]>
[2024-04-29T14:04:52.503+0000] {taskinstance.py:2193} INFO - Starting attempt 4 of 4
[2024-04-29T14:04:52.528+0000] {taskinstance.py:2217} INFO - Executing <Mapped(_PythonDecoratedOperator): ETL_job> on 2024-04-29 13:40:00+00:00
[2024-04-29T14:04:52.536+0000] {standard_task_runner.py:60} INFO - Started process 639944 to run task
[2024-04-29T14:04:52.545+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'Units_statistics_DAG', 'ETL_job', 'scheduled__2024-04-29T13:40:00+00:00', '--job-id', '23130', '--raw', '--subdir', 'DAGS_FOLDER/Units_dag.py', '--cfg-path', '/tmp/tmp77rkopvr', '--map-index', '16']
[2024-04-29T14:04:52.547+0000] {standard_task_runner.py:88} INFO - Job 23130: Subtask ETL_job
[2024-04-29T14:04:52.637+0000] {task_command.py:423} INFO - Running <TaskInstance: Units_statistics_DAG.ETL_job scheduled__2024-04-29T13:40:00+00:00 map_index=16 [running]> on host a09da322e808
[2024-04-29T14:04:52.826+0000] {taskinstance.py:2513} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='abeer-araby' AIRFLOW_CTX_DAG_ID='Units_statistics_DAG' AIRFLOW_CTX_TASK_ID='ETL_job' AIRFLOW_CTX_EXECUTION_DATE='2024-04-29T13:40:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-04-29T13:40:00+00:00'
[2024-04-29T14:04:52.828+0000] {Units_dag.py:99} INFO - i: 160000, chunk: 8725
[2024-04-29T14:04:52.828+0000] {Units_dag.py:100} INFO - child token is::: eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczovL2FwaS5hZmFxeS5zYS9hdXRoL2xvZ2luX2FzIiwiaWF0IjoxNzE0Mzk5MjA0LCJleHAiOjE3MTY5OTEyMDQsIm5iZiI6MTcxNDM5OTIwNCwianRpIjoiV2JUNnU0UUZWWk50Z3Q5TiIsInN1YiI6IjVkNjNlNTFlMDNlZDUyOTJhNDJjMjRlNiIsInBydiI6IjI3MGVjZmM3ZWEzZWQ5MzdlYTg0OTM2MmEzYTUwOTEwYzZkOGNlNGYiLCJsb2dpbl9hc19pZCI6IjY1N2YwYWQ3M2Y5NzcyOTVjZjA1NTc2YSJ9.tVP0N8r6XYfHHbyxyguTZtK0S7EUhhtlsm4GjqIITis
[2024-04-29T14:05:09.073+0000] {Units_dag.py:104} INFO - {'_id': '65ec4b3bd86150e572013169', 'icon': 'icons/P_4.png', 'driver_id': '65fcd370e0a54f7a630f0da2', 'tailer_id': '', 'profile': {'vehicle_type': '', 'measurement_type': 'length', 'max_capacity': 0, 'vin': '', 'plate_number': '', 'start_operation': '', 'contract_milage': '', 'opening_milage': '', 'model': '', 'fuel_type': '91', 'year': 0, 'seats': 2}, 'shipment_info': {'product_category': None, 'truck_type': None, 'shipment_type': None, 'shipment_payment_type': None, 'transporter_type': None, 'operating_area': None, 'weight_kilograms_per_day': None, 'kms_per_day': None, 'shipment_trip_drive_hrs_day': None, 'allowable_delay_drive_hrs_day': None, 'expected_shipment_trip_day': None, 'car_pay_per_day': None, 'exceeding_temperature_limit_per_day': None, 'temperature_upper_limit': None, 'exceeding_humidity_limit_per_day': None, 'humidity_upper_limit': None}, 'device_password': '***', 'device_type': 'gps', 'gateway': False, 'business_types': {'waste': False, 'transportation': True, 'water': False}, 'name': '1101AD', 'operation_code': '', 'device_gateway': '', 'skip_sim_information': False, 'imei': '866069062284516', 'sim_number': '+966830020558434', 'sim_serial': '8996609821112199408', 'device_serial': '1135867115', 'device': 'teltonika', 'user_id': '65d61c06eb33bd22e909fc25', 'created_by': '5e6655bf03ed52e1ab79d301', 'updated_at': '2024-04-18 11:18:59', 'created_at': '2024-03-09 14:42:51', 'custom_fields': [], 'creator': 'amr.ahmed', 'company': '', 'owner': 'debrah-logistics-admin', 'last_update': {'dtt': 1714399239000, 'dts': 1714399241715, 'spd': 0, 'ang': 276, 'alt': 4, 'chPrams': {'Speed': {'v': 0, 'cdt': 1714337121000}, 'output1': {'v': False, 'cdt': 1714347246000}, 'ePwrV': {'v': 12.401, 'cdt': 1714399239000}, 'parm24': {'v': 0, 'cdt': 1714337121000}, 'sat': {'v': 13, 'cdt': 1714398637000}, 'ai1': {'v': 174, 'cdt': 1714384512000}, 'type': {'v': 6, 'cdt': 1712923799000}, 'priority': {'v': 0, 'cdt': 1714384512000}, 'di1': {'v': 0, 'cdt': 1714337151000}, 'result': {'v': '5665723a30332e32382e30325f313937204750533a41584e5f352e312e392048773a464d43393230204d6f643a3120494d45493a38363630363930363232383435313620496e69743a323032342d332d313320383a333420557074696d653a34303231353038204d41433a333838413231393232463833205350433a312830292041584c3a31204f42443a3020424c3a312e31302042543a34', 'cdt': 1714346945000}, 'protocol': {'v': 'FMC920', 'cdt': 1714347246000}, 'iPwrV': {'v': 3.963, 'cdt': 1714389922000}, 'event': {'v': 0, 'cdt': 1714386616000}}, 'prms': {'protocol': 'FMC920', 'priority': 0, 'sat': 13, 'event': 0, 'di1': 0, 'output1': 0, 'ePwrV': 12.401, 'Speed': 0, 'parm24': 0, 'iPwrV': 3.963, 'ai1': 174}, 'devPrms': ['protocol', 'priority', 'sat', 'event', 'di1', 'output1', 'ePwrV', 'Speed', 'parm24', 'iPwrV', 'ai1'], 'old_params': {'Speed': 0, 'protocol': 'teltonika', 'output1': 0, 'ePwrV': 12.392, 'parm24': 0, 'sat': 13, 'ai1': 174, 'iPwrV': 3.963, 'priority': 0, 'event': 0, 'di1': 0}, 'vLoc': 1, 'vCon': 1, 'ip': '109.83.230.21', 'rl': 0, 'trl': 8295.4344, 'mdu': 0, 'tmdu': 665805, 'eidl': 0, 'teidl': 223229, 'fc': 0, 'tfc': 995.452128, 'acc': 0, 'accOn': 2091740, 'accOff': 3107544, 'sdu': 301, 'tsdu': 62088, 'lat': 21.5614366, 'lng': 39.1370866}, 'counters': {'engine_hours': 2091740, 'last_acc': 0, 'odometer': 8294.5017, 'engine_hours_type': 'acc', 'odometer_type': 'gps'}, 'is_wasl_connected': 'Not Connected', 'iconUrl': 'https://api.afaqy.sa/images/units/icons/P_4.png', 'hasCCTV': False}
[2024-04-29T14:05:09.597+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/elasticsearch/_sync/client/__init__.py:399: SecurityWarning: Connecting to 'https://dbelk3.afaqy.local:9200' using TLS with verify_certs=False is insecure
  _transport = transport_class(

[2024-04-29T14:05:09.649+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/elasticsearch/_sync/client/__init__.py:399: SecurityWarning: Connecting to 'https://dbelk1.afaqy.local:9200' using TLS with verify_certs=False is insecure
  _transport = transport_class(

[2024-04-29T14:05:09.698+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/elasticsearch/_sync/client/__init__.py:399: SecurityWarning: Connecting to 'https://dbelk2.afaqy.local:9200' using TLS with verify_certs=False is insecure
  _transport = transport_class(

[2024-04-29T14:05:09.719+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host 'dbelk3.afaqy.local'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(

[2024-04-29T14:05:09.793+0000] {_transport.py:335} INFO - GET https://dbelk3.afaqy.local:9200/_cluster/health [status:200 duration:0.093s]
[2024-04-29T14:05:10.178+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/urllib3/connectionpool.py:1061: InsecureRequestWarning: Unverified HTTPS request is being made to host 'dbelk1.afaqy.local'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/1.26.x/advanced-usage.html#ssl-warnings
  warnings.warn(

[2024-04-29T14:05:11.280+0000] {_transport.py:335} INFO - PUT https://dbelk1.afaqy.local:9200/units_statistics_***/_bulk [status:200 duration:1.124s]
[2024-04-29T14:05:11.370+0000] {taskinstance.py:2731} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/decorators/base.py", line 241, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 200, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 217, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/Units_dag.py", line 112, in ETL_job
    Elk_client.elk_pushData(Globals.index_name,units_list_update)
  File "/opt/airflow/dags/elk_manager.py", line 69, in elk_pushData
    helpers.bulk(self.elk_client, docs_list, index=index_name, chunk_size=batch)
  File "/home/airflow/.local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 521, in bulk
    for ok, item in streaming_bulk(
  File "/home/airflow/.local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 436, in streaming_bulk
    for data, (ok, info) in zip(
  File "/home/airflow/.local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 355, in _process_bulk_chunk
    yield from gen
  File "/home/airflow/.local/lib/python3.8/site-packages/elasticsearch/helpers/actions.py", line 274, in _process_bulk_chunk_success
    raise BulkIndexError(f"{len(errors)} document(s) failed to index.", errors)
elasticsearch.helpers.BulkIndexError: 88 document(s) failed to index.
[2024-04-29T14:05:11.386+0000] {taskinstance.py:1149} INFO - Marking task as FAILED. dag_id=Units_statistics_DAG, task_id=ETL_job, map_index=16, execution_date=20240429T134000, start_date=20240429T140452, end_date=20240429T140511
[2024-04-29T14:05:11.412+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 23130 for task ETL_job (88 document(s) failed to index.; 639944)
[2024-04-29T14:05:11.463+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-04-29T14:05:11.494+0000] {taskinstance.py:3312} INFO - 0 downstream tasks scheduled from follow-on schedule check
